runtime/src/main/scala/akka/grpc/netty/NettyCircuitBreakerInterceptor.scala

```diff
@@ -0,0 +1,57 @@
+package akka.grpc.netty
+
+import akka.pattern.CircuitBreaker
+import io.grpc.CallOptions
+import io.grpc.Channel
+import io.grpc.ClientCall
+import io.grpc.ClientInterceptor
+import io.grpc.ClientInterceptors
+import io.grpc.ForwardingClientCallListener
+import io.grpc.Metadata
+import io.grpc.MethodDescriptor
+import io.grpc.Status
+
+/**
+ * A `io.grpc.ClientInterceptor` that prevents downstream overload wrapping the calls with a circuit breaker.
+ */
+class NettyCircuitBreakerInterceptor(breaker: CircuitBreaker) extends ClientInterceptor {
```

Comment:
This [interceptor](https://github.com/grpc/grpc-java/blob/master/api/src/main/java/io/grpc/ClientInterceptor.java#L21-L35) would be used like this:

```
val circuitBreakingInterceptor = new NettyCircuitBreakerInterceptor(myBreaker)
GrpcClientSettings.fromConfig("helloworld.GreeterService")
      .withChannelBuilderOverrides{ builder =>
        builder.intercept(circuitBreakingInterceptor)
      }
```

so even if the interceptor seems to operate at call level (only `interceptCall()` is provided), I think there's a single instance per client. 

I am not sure how an Interceptor interferes with the Channel lifecycle.